src/libs/ImplicitAuthManager.js

```diff
@@ -0,0 +1,537 @@
+ *   realmName: {string} [required] name of your realm within the domain,
+ *   redirectURI: {string} | {function} [optional] defaults to window.location.origin + intention
+ *   // if redirect URI is a string, during login/logout processes the redirectURI will have a query param appended to it
+ *   // based on the intention of the process: ie if the call to the authorization server is intended for logging in
+ *   // the redirect uri will = redirect_uri + ?intention=LOGIN
+ *   // --
+ *   // you may make the redirect URI a call back function that receives the intention as an argument
+ *   // view the validAPIIntentions static function for all the possible intentions that you may want to handle
+ *   loginURIResponseType: {string} [optional] defaults to id_token (options are token | id_token | id_token token)
+ *   // please view https://stackoverflow.com/questions/19293793/oauth-2-access-token-vs-openid-connect-id-token
+ *   // for clarification on the difference between token types
+ *   hooks: {
+ *      onBeforeAuthRedirect: {function},
+ *      onAfterAuthRedirect: {function} [error {boolean}],
+ *      onAuthLocalStorageCleared: {function},
+ *      onTokenExpired: {function},
+ *      onAuthenticateSuccess: {function},
+ *      onAuthenticateFail: {function}
+ *   }
+ * }
+ */
+
+// ImplicitAuthManager takes two names in the local storage space
+// sso and auth. Please ensure these are not being overwritten by other functions.
+export default class ImplicitAuthManager {
+  constructor(config = {}) {
+    //default config
+    const defaultConfig = this.defaultConfig;
+    //validate config
+    this.validateConfig(config);
+    //merge defaults with config
+    this.config = {
+      ...defaultConfig,
+      ...config,
+      hooks: { ...defaultConfig.hooks, ...config.hooks },
+    };
+    this.baseAuthEndpoint = this.createBaseAuthEndpointFromConfig();
+    this.baseLogoutEndpoint = this.createBaseLogoutEndpointFromConfig();
+  }
+
+  get defaultConfig() {
+    return {
+      loginURIResponseType: 'id_token',
+      hooks: {
+        onBeforeAuthRedirect: () => undefined,
+        onAfterAuthRedirect: () => undefined,
+        onAuthLocalStorageCleared: () => undefined,
+        onTokenExpired: () => undefined,
+        onAuthenticateSuccess: () => undefined,
+        onAuthenticateFail: () => undefined,
+      },
+    };
+  }
+
+  get hooks() {
+    return this.config.hooks;
+  }
+  get redirectURI() {
+    return this.config.redirectURI || window.location.origin;
+  }
+
+  get ssoLogoutURI() {
+    return this.getSSOLogoutURI();
+  }
+
+  get ssoLoginURI() {
+    return this.getSSOLoginURI();
+  }
+
+  get access_token() {
+    return this.getAccessTokenFromLocal();
+  }
+
+  get id_token() {
+    return this.getIdTokenFromLocal();
+  }
+  //returns the valid response types for implicit auth flow response_type query param
+  static validResponseTypes() {
+    return ['id_token', 'token', 'id_token token'];
```

Comment:
Is the space meant to be in `id_token token`?

Reply:
yep! It gets encoded later on to be id_token%20token